src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -1638,6 +1638,9 @@ interface I
                 // (5,26): error CS8701: Target runtime doesn't support default interface implementation.
                 //     public static object F2;
                 Diagnostic(ErrorCode.ERR_RuntimeDoesNotSupportDefaultInterfaceImplementation, "F2").WithLocation(5, 26),
+                // (5,26): warning CS8618: Non-nullable field 'F2' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.
```

Comment:
I'm not sure we care about this warning, since there's an error on F2 already.
Even with a runtime that supports DIM, we'd get an error because interfaces cannot contain state.
So I'm tempted to say we shouldn't make this change at all.

Reply:
Static fields are allowed on interfaces when the runtime supports DIM.

Reply:
> I'm not sure we care about this warning, since there's an error on F2 already.

The warning on F2 is completely different in nature and we generally accept both warnings being given simultaneously. Such a warning would be removed if the member is written anywhere in the program. The nullable init warning would only be removed by adding an initializer or static constructor which initializes the member.

Reply:
I would also be happy to just make the interface public in this test in order to reduce the noise of the uninitialized field warnings.

Reply:
Thanks much for the clarification. I'd forgotten that we allowed static fields (and constructors) in interfaces.

---
In reply to: [472494540](https://github.com/dotnet/roslyn/pull/46928#discussion_r472494540) [](ancestors = 472494540)